### PR TITLE
Rewrite Part 2

### PR DESCRIPTION
**Pull request by shortdudey123**
- Mainly updating the user management to work w/ 2.x mongo gem as well as the existing code for 1.x
- Cleanup all deprecation warnings as well

Closes https://github.com/sous-chefs/mongodb/issues/119

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,6 +15,7 @@ install: echo "skip bundle install"
 branches:
   only:
     - master
+    - rewrite_v1
```
Comment (damacus) [inline .travis.yml]: still needed?

File: test/integration/user_management_v2/inspec/user_management_spec.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# @test "requires authentication" {
```
Comment (damacus) [inline test/integration/user_management_v2/inspec/user_management_spec.rb]: probably worth deleting now?

File: .travis.yml (hunk 1 above)
Comment (shortdudey123) [inline .travis.yml]: yes since i am merging all of the rewrite parts into this branch
otherwise travis tests don't run on my PR's :)

Comment (shortdudey123) [inline .travis.yml]: Will remove it after the `rewrite_v1` is merged to master

File: test/integration/user_management_v2/inspec/user_management_spec.rb (hunk 2 above)
Comment (shortdudey123) [inline test/integration/user_management_v2/inspec/user_management_spec.rb]: probably :)
Will yank it in part 3

**Pull request merged**